src/ucp/am/eager_multi.c

```diff
@@ -117,11 +117,14 @@ static UCS_F_ALWAYS_INLINE ucs_status_t ucp_am_eager_multi_bcopy_send_func(
         ucp_request_t *req, const ucp_proto_multi_lane_priv_t *lpriv,
         ucp_datatype_iter_t *next_iter)
 {
+    size_t user_hdr_size = req->send.msg_proto.am.header_length;
+
     return ucp_proto_eager_bcopy_multi_common_send_func(
             req, lpriv, next_iter, UCP_AM_ID_AM_FIRST,
             ucp_am_eager_multi_bcopy_pack_args_first,
-            UCP_AM_FIRST_FRAG_META_LEN, UCP_AM_ID_AM_MIDDLE,
-            ucp_am_eager_multi_bcopy_pack_args_mid, UCP_AM_MID_FRAG_META_LEN);
+            UCP_AM_FIRST_FRAG_META_LEN + user_hdr_size, UCP_AM_ID_AM_MIDDLE,
```

Comment:
it seems weird that we are passing user_hdr_size to ucp_proto_eager_bcopy_multi_common_send_func, but actually this function does not really pack the header; then we add the header size inside ucp_am_eager_multi_bcopy_pack_args_first to compensate to subtracting it..
firstly, ucp_am_eager_bcopy_pack_data should take max_length without the header. maybe even rename it to ucp_am_eager_bcopy_pack